kaggle/api/kaggle_api_extended.py

```diff
@@ -763,14 +768,16 @@ def competitions_list(
         category: category to filter result to; use 'all' to get closed competitions
         group: group to filter result to
         """
+        group_val = CompetitionListTab.COMPETITION_LIST_TAB_EVERYTHING
         if group:
             if group not in self.valid_competition_groups:
                 raise ValueError('Invalid group specified. Valid options are ' + str(self.valid_competition_groups))
             if group == 'all':
-                group = CompetitionListTab.COMPETITION_LIST_TAB_EVERYTHING
+                group_val = CompetitionListTab.COMPETITION_LIST_TAB_EVERYTHING
             else:
-                group = self.lookup_enum(CompetitionListTab, group)
+                group_val = cast(CompetitionListTab, self.lookup_enum(CompetitionListTab, group))
```

Comment:
Can `lookup_enum` return a type based on the type of it's first argument? Similar to generics in Java. This may solve the `cast` issue but I am not sure that's possible.

Reply:
I tried to make that happen but was not able to figure it out. Since then, I have seen a reference to generics in the `mypy` docs, so I'll try again.

Reply:
It looks like it could be made to work, if we switch to 3.12. The TypeVar syntax isn't available in prior versions. I don't think we can make that switch, but I'm going to see if I can do it anyway, as an exercise.

Reply:
I could not find a way to specify that the return type was an instance of a class-type parameter. I ended up adding an additional parameter to the method, which is a sample of the correct type. Looks like this:
```python
  request.framework = self.lookup_enum(ModelFramework, ModelFramework.MODEL_FRAMEWORK_API, framework)
```
Not ideal, but it will do.

With a little work in the code that prompts users to upgrade, it seems like we can switch to requiring Python 3.12.